csrc/transform_replay.cpp

```diff
@@ -233,6 +233,97 @@ TensorDomain* TransformReplay::fullSelfReplay(
       new_self_root->contiguity());
 }
 
+TensorDomain* TransformReplay::selfAllocationReplay(
```

Comment:
I tagged @wujingyue because I think you asked for something similar to this?

Though I'm not totally sure if this does what you want, because I'm addressing some requirement from `remove_bcast_squeeze`, which I think we can also refactor this one and put some logic on the user side, if we needed a more generic replay later.